controllers/drplacementcontrol.go

```diff
@@ -1453,16 +1453,8 @@ func (d *DRPCInstance) createVRGManifestWork(homeCluster string, repState rmn.Re
 	return nil
 }
 
+// ensureVRGManifestWork ensures that the VRG manifestwork exists and matches the current VRG state.
```

Comment:
We now update MW for VRG from the cached MCV VRG replicationState when this function is invoked.

While this is not a problem when called from `ensureActionCompleted` where its callers ensure that the cached VRG is primary and we then invoke `ensureVRGManifestWork`.

There are some gaps in the invocation from `RunInitialDeployment` that needs closer inspection.

For example the code that decides a VRG exists on one of the clusters is not checking if that VRG exists as primary, if that was done using the replication state from the MCV is fine, as it would always be primary when this function is called.

This function `ensureVRGManifestWork` is currently only safe when VRG MCV is Primary and desired VRG MW is also Primary.

Reply:
Is this issue new? this changes only calls when the vrg exists - this could be called for the wrong vrg state before this change if the vrg manifest work was deleted, no?